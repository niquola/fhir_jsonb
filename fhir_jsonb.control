# jsquery extension
comment = 'tests for json b'
default_version = '1.0'
module_pathname = '$libdir/fhir_jsonb'
relocatable = true

