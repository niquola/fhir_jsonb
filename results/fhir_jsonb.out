CREATE EXTENSION fhir_jsonb;
CREATE EXTENSION jsquery;
SET escape_string_warning=off;
SELECT count(*)
FROM patients;
 count 
-------
   100
(1 row)

SELECT DISTINCT ON (doc #>> '{active}')
doc #>> '{active}'
FROM patients
ORDER BY doc #>> '{active}';
 ?column? 
----------
 false
 true
(2 rows)

SELECT count(*)
FROM encounters;
 count 
-------
  1000
(1 row)

SELECT DISTINCT ON (doc #>> '{status}')
doc #>> '{status}'
FROM encounters
ORDER BY doc #>> '{status}';
  ?column?   
-------------
 cancelled
 finished
 in progress
 onleave
 planned
(5 rows)

SELECT DISTINCT ON (doc #>> '{class}')
doc #>> '{class}'
FROM encounters
ORDER BY doc #>> '{class}';
 ?column?  
-----------
 emergency
 inpatient
(2 rows)

SELECT DISTINCT ON (doc #>> '{participant,0,individual,type}')
doc #>> '{participant,0,individual,type,0,coding,0,code}'
FROM encounters
ORDER BY doc #>> '{participant,0,individual,type}';
 ?column? 
----------
 ADM
 ATND
 CALLBCK
 CON
 DIS
 ESC
 REF
(7 rows)

SELECT DISTINCT ON (doc #>> '{participant,0,individual,reference}')
doc #>> '{participant,0,individual,reference}'
FROM encounters
ORDER BY doc #>> '{participant,0,individual,reference}';
         ?column?         
--------------------------
 Alexander Fleming
 Charles R. Drew
 Christiaan Eijkman
 Daniel Carleton Gajdusek
 Edward Jenner
 Elliott P. Joslin
 Ernst Haeckel
 Galen
 Garcia de Orta
 George E. Goodfellow
 Girolamo Fracastoro
 Helen Flanders Dunbar
 Henry Gray
 Henry Heimlich
 Hippocrates
 Ian Olver
 John Hunter
 Orvan Hess
 Pierre Fauchard
 Rene Geronimo Favaloro
 Sigmund Freud
 William Harvey
(22 rows)

SELECT DISTINCT ON (doc #>> '{reason}')
doc #>> '{reason,text}'
FROM encounters
ORDER BY doc #>> '{reason}';
                         ?column?                          
-----------------------------------------------------------
 f asdfsd fasdfsad fdasfklsdjaflksda fjsdafjasdklf jsdklfj
 fdaff asdf asdfasdf sdaf asdfsd fasdfsad fdasfk
 lsdjaflksda fjsdafjasdklf jsdklfj
(3 rows)

-- SELECT
--   doc#>'{participant,0}' as part,
--   doc#>'{class}' as cls,
--   doc#>'{status}' as status
-- FROM encounters
-- WHERE
-- doc @@
-- '("class" = "emergency" &
--   "participant".#."individual" (
--     "reference" = "Galen" &
--     "type".#."coding".# (
--       "code" && ["ADM", "ATND"] &
--       "system" = "encounter-participant-type")) &
--   !("status" && ["planned", "finished", "cancelled"]))'
-- LIMIT 10;
SELECT count(*) > 0
FROM encounters
WHERE
doc @@
'("class" = "emergency" &
  "participant".#."individual" (
    "reference" = "Galen" &
    "type".#."coding".# (
      "code" in ("ADM","ATND") &
      "system" = "encounter-participant-type")) &
  !("status" && ["planned", "finished", "cancelled"]))';
 ?column? 
----------
 t
(1 row)

-- SELECT count(*)
-- FROM encounters
-- WHERE
-- doc @@
-- '("class" = "emergency" &
--   "participant".#."individual" (
--     "reference" = "Galen" &
--     "type".#."coding".# (
--       "code" && ["ADM", "ATND"] &
--       "system" = "encounter-participant-type")) &
--   !("status" && ["planned", "finished", "cancelled"]))';
SELECT count(*)
FROM observations;
 count 
-------
 10000
(1 row)

SELECT DISTINCT ON (doc #>> '{name}')
doc #>> '{name,coding,0,display}'
FROM observations
ORDER BY doc #>> '{name}';
              ?column?               
-------------------------------------
 Blood pressure systolic & diastolic
 Body temperature
 noise
(3 rows)

SELECT DISTINCT ON (doc #>> '{status}')
doc #>> '{status}'
FROM observations
ORDER BY doc #>> '{status}';
     ?column?     
------------------
 amended
 cancelled
 entered in error
 final
 preliminary
 registered
(6 rows)

-- SELECT count(*) FROM observations
-- WHERE doc @@ '(
--   "name"."coding".# (
--     "system" = "http://loinc.org"
--      &
--     "code" = "8310-5"
--   )
-- )'
-- AND
-- (doc #>> '{appliesPeriod,start}')::timestamp > CURRENT_DATE - INTERVAL '3 month'
-- LIMIT 10;
SELECT count(*)
FROM conditions;
 count 
-------
 10000
(1 row)

SELECT DISTINCT ON (doc #>> '{code}')
doc #>> '{code,coding,0,code}'
FROM conditions
ORDER BY doc #>> '{code}';
 ?column? 
----------
 430
 431
 433.01
 433.10
 433.11
 433.21
 433.31
 433.81
 433.91
 434.00
 434.01
 434.11
 434.91
 436
 noise
(15 rows)

SELECT DISTINCT ON (doc #>> '{category}')
doc #>> '{category,coding,0,code}'
FROM conditions
ORDER BY doc #>> '{category}';
 ?column?  
-----------
 complaint
 diagnosis
 finding
 symptom
(4 rows)

SELECT DISTINCT ON (doc #>> '{status}')
doc #>> '{status}'
FROM conditions
ORDER BY doc #>> '{status}';
  ?column?   
-------------
 confirmed
 provisional
 refuted
 working
(4 rows)

-- SELECT
-- doc#>'{status}',
-- doc#>'{category,coding}',
-- doc#>'{code,coding}'
-- FROM conditions
-- LIMIT 10;
-- SELECT count(*)
-- FROM conditions
-- WHERE
-- doc @@
-- '("status" = "confirmed" &
--   "category"."coding".# (
--     "system" = "http://snomed.info/sct" &
--     "code" = "diagnosis") &
--   "code"."coding".#."system" = "http://snomed.info/sct")'
-- AND (doc #>> '{code,coding,0,code}' ) in  (SELECT code from stroke_diagnoses) ;
-- SELECT count(*)
-- FROM conditions
-- WHERE
-- doc @@
-- '("status" = "confirmed" &
--   "category"."coding".# (
--     "system" = "http://snomed.info/sct" &
--     "code" = "diagnosis") &
--   "code"."coding".# (
--     "system" = "http://snomed.info/sct" &
--     "code" in ("433.01", "433.10", "433.11", "433.21", "433.31", "433.81", "433.91", "434.00", "434.01", "434.11", "434.91", "436", "430", "431")
--   ))'
-- ;
