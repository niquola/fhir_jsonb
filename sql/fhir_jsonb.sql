CREATE EXTENSION fhir_jsonb;
set escape_string_warning=off;

select count(*)
from patients;

select count(*)
from encounters;

select count(*)
from observations;
