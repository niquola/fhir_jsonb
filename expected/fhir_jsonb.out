CREATE EXTENSION fhir_jsonb;
set escape_string_warning=off;
select count(*)
from patients;
 count 
-------
   100
(1 row)

select count(*)
from encounters;
 count 
-------
  1000
(1 row)

select count(*)
from observations;
 count 
-------
 10000
(1 row)

